### ISSUE TITLE
Add non-Qt to Travis builds

### ISSUE THREAD

**Issue by vivid-synth** [supercollider/supercollider#2306]
Ideally the first test so we can "fail fast"

**Comment by vivid-synth** [comment]
I don't know how to do this (maybe can look into it soon), but in general we just want to run a separate test which is the same but doesn't export ```QT_PREFIX``` or ```QT_PLUGIN_PATH```, and which has the flags ```-DSC_IDE=False -DSC_QT=False```

**Comment by gusano** [comment]
@vivid-synth I'll be up for tackling that one but I'd rather wait for https://github.com/supercollider/supercollider/pull/2298 to be merged into master first ;)

**Comment by scztt** [comment]
It should be possible (trivially easy?) to run parallel travis.ci jobs, one for SC_QT=True and one for SC_QT=False. I believe you specify the various options in the env block of the yaml, and it'll generate jobs for each combination [https://docs.travis-ci.com/user/customizing-the-build#Build-Matrix]. 

I would suggest experimenting with that in a branch to make sure the jobs and parameters are getting generated okay, so it won't interfere with regular travis status. I think we'd probably need to have two versions of travis_test_run_proto.json, one including UI tests and one w/o UI tests as well. But, this shouldn't really be too tied up with #2298, as it's probably entails only travis-related changes. I'd be glad to help out w/ setting this up if needed, let me know.

**Comment by gusano** [comment]
@scztt yeah setting up the build matrix is easy.
It's just I moved the linting in a separate script in the other PR and it might be just easier to start cleaning travis things up from there..
Just my 2 cents anyway.

Not sure if that would make sense to add more builds to the matrix (re `travis_test_run_proto.json`) as our builds already take ages.

**Comment by scztt** [comment]
As far as I know, builds will be run in parallel, so additional builds in the matrix won't increase the time it takes to determine pass/fail status. And, the full test suite (```travis_test_run_proto.json``` is everything except server tests) is tiny compared to the build time, less than a minute I think, so we could easily load up more tests without any serious consequences.

I agree 100% about breaking out some of the shell commands in the travis.yml into separate scripts. It reached critical mass a while ago, but also reached a fairly stable point in terms of what it was doing. I don't think it's wise to cram in any more functionality without refactoring a bit.

**Comment by gusano** [comment]
@scztt I was playing with it a bit and the 4 builds were taking up to 1h (parallel builds)..
No time right now but I'll submit a WIP PR in the next days if no one else does.

**Comment by scztt** [comment]
@gusano Ah, looking at it now - the real slowpoke is linux. It takes longer to install QT via apt then it does to actually build SuperCollider. MacOS only takes ~15 minutes to build and test. This could be fixed by going back to a fixed, download-and-unzip distrib of QT for linux builds (and accepting the various consequences of that).

Honestly, given our volume of commits (e.g. a few a day) and the fact that PR's really should really bake for a few days to a week for review _anyway_, sub-15-minute build response times seems like a nice-to-have but not essential. We could get that 15 minute build time down to <5 by caching build intermediates between travis runs (which isn't super hard), but we'd be (slightly?) increasing the chance of cruft-related breakages.

**Comment by scztt** [comment]
Hahah, wait....? After switching to apt-get QT installation, linux went from 15m to 30m. But somewhere since then, linux went back to 15m and now MAC is 30m.... Now I'm confused - though, nonetheless, I think the CORRECT clean build+test time should be around 15 minutes, and anything more than that should be considered worth investigation....

**Comment by gusano** [comment]
@scztt unrelated question: when building with `-DSC_QT=OFF -DSC_IDE=OFF` I get the following error:

```shell
/home/gusi/dev/supercollider/common/localsocket_utils.hpp:4:22: fatal error:
QByteArray: No such file or directory
#include <QByteArray>
```

Any idea?

**Comment by vivid-synth** [comment]
Done, thanks @gusano!
